portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,19 +57,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numCommentDisplay = getNumDisplay(request);
+    if (numCommentDisplay == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 1 and 10.");
+      return;
+    }
```

Comment:
You don't have to do this in this cl, but it'd be nice to have some error message on the website to indicate that the user's entered value is invalid.

Reply:
One side-effect of the current error-handling is that the response sometimes is plain text and is sometimes JSON, and since your client-side JS expects JSON, it crashes when this plain text is sent back. You can repro by blanking out the comment number and pressing "Display".

But as Joan said, that doesn't have to be changed in this PR.